whois-api/src/test/java/net/ripe/db/whois/api/nrtmv4/NrtmClientServiceTestIntegration.java

```diff
@@ -261,10 +272,57 @@ public void should_get_update_notification_file() {
     }
 
     @Test
-    public void should_get_delta_file() throws JSONException {
+    public void should_get_signature_file() throws DecoderException {
+        insertUpdateNotificationFile();
+        generateKeyPair();
 
-        snapshotFileGenerator.createSnapshot();
+        final String notificationFile = getResponseFromHttpsRequest("TEST/update-notification-file.json", MediaType.APPLICATION_JSON).readEntity(String.class);
+
+        final Response response = getResponseFromHttpsRequest("TEST/update-notification-file.json.sig", MediaType.APPLICATION_JSON);
+        assertThat(response.getStatus(), is(200));
+        final String signature = response.readEntity(String.class);
+
+        // verify the signature
+        Signer verifier = new Ed25519Signer();
+
+        verifier.init(false, new Ed25519PublicKeyParameters(nrtmKeyConfigDao.getPublicKey(), 0));
+        verifier.update(notificationFile.getBytes(), 0, notificationFile.getBytes().length);
+        assertThat(verifier.verifySignature(Hex.decodeHex(signature)), is(Boolean.TRUE));
+    }
+
+    @Test
+    public void should_fail_to_verify_signature_file() throws DecoderException {
+        insertUpdateNotificationFile();
+        generateKeyPair();
+
+        final String notificationFile = getResponseFromHttpsRequest("TEST/update-notification-file.json", MediaType.APPLICATION_JSON).readEntity(String.class);
+
+        final Response response = getResponseFromHttpsRequest("TEST/update-notification-file.json.sig", MediaType.APPLICATION_JSON);
+        assertThat(response.getStatus(), is(200));
+        assertThat(response.getHeaderString(HttpHeaders.CACHE_CONTROL), is("public, max-age=60"));
 
+        final String signature = response.readEntity(String.class);
+
+        // verify the signature
+        Signer verifier = new Ed25519Signer();
+
+        verifier.init(false, generateKeyPair().getPublic());
+        verifier.update(notificationFile.getBytes(), 0, notificationFile.getBytes().length);
+        assertThat(verifier.verifySignature(Hex.decodeHex(signature)), is(Boolean.FALSE));
+    }
+
+    @Test
+    public void should_throw_exeption_no_key_exists() {
+        insertUpdateNotificationFile();
+
+        final Response response = getResponseFromHttpsRequest("TEST/update-notification-file.json.sig", MediaType.APPLICATION_JSON);
```

Comment:
Use a String entity instead of response and use an `assertThrows` an `InternalErrorException` instead?